src/passes/functionPruner/callGraph.ts

```diff
@@ -0,0 +1,72 @@
+import assert from 'assert';
+import {
+  Expression,
+  FunctionCall,
+  FunctionDefinition,
+  Identifier,
+  IdentifierPath,
+  MemberAccess,
+} from 'solc-typed-ast';
+import { AST } from '../../ast/ast';
+import { ASTMapper } from '../../ast/mapper';
+import { printNode } from '../../utils/astPrinter';
+
+export class CallGraphBuilder extends ASTMapper {
+  callGraph: Map<number, Set<number>>;
+  functionId: Map<number, FunctionDefinition>;
+  currentFunction: FunctionDefinition | undefined;
+
+  constructor() {
+    super();
+    this.callGraph = new Map();
+    this.functionId = new Map();
+    this.currentFunction = undefined;
+  }
+
+  visitFunctionDefinition(node: FunctionDefinition, ast: AST) {
+    this.currentFunction = node;
+    this.functionId.set(node.id, node);
+    this.callGraph.set(node.id, new Set());
+    this.commonVisit(node, ast);
+    this.currentFunction = undefined;
+  }
+
+  visitFunctionCall(node: FunctionCall, ast: AST) {
+    assert(
+      this.currentFunction !== undefined,
+      `Function Call ${printNode(node)} outside FunctionDefinition`,
+    );
+    const existingCalls = this.callGraph.get(this.currentFunction.id);
+    assert(
+      existingCalls !== undefined,
+      `${printNode(this.currentFunction)} should have been added to the map`,
+    );
+
+    const refFuncId = getReferencedFunctionId(node.vExpression);
```

Comment:
FunctionCall already has a vReferencedDeclaration member, any reason for having a custom function rather than using that?